Added pipeline job to load f_user_course in vertica.

HassanJaveed84 opened the pull request:
@mulby @brianhw

HassanJaveed84 commented on edx/analytics/tasks/load_internal_reporting_user_course.py:
```diff
@@ -0,0 +1,53 @@
+        """The schema has enrollment_is_active as well, but 'course_enrollment' hive table does not have it."""
+        return [
+            ('user_id', 'INTEGER NOT NULL'),
+            ('course_id', 'VARCHAR(200) NOT NULL'),
```
Should we use varchar(256) here ?

mulby replied:
Yes

brianhw commented on edx/analytics/tasks/load_internal_reporting_user_course.py:
```diff
@@ -0,0 +1,53 @@
+
+
+class LoadInternalReportingUserCourseToWarehouse(WarehouseMixin, VerticaCopyTask):
+
```
nit: docstring.

HassanJaveed84 replied:
Done.

brianhw commented on edx/analytics/tasks/load_internal_reporting_user_course.py:
```diff
@@ -0,0 +1,53 @@
+
+    @property
+    def table(self):
+        return 'f_user_course_temp'
```
Is this a temporary name?  The acceptance test just looks for "f_user_course", so I assume so.

HassanJaveed84 replied:
Yes, fixed.

brianhw commented on edx/analytics/tasks/tests/acceptance/test_internal_reporting_user_course.py:
```diff
@@ -0,0 +1,53 @@
+            expected_output_csv = os.path.join(self.data_dir, 'output', 'acceptance_expected_f_user_course.csv')
+            expected = pandas.read_csv(expected_output_csv, parse_dates=True)
+
+            cursor.execute("SELECT * FROM {schema}.f_user_course".format(schema=self.vertica.schema_name))
```
This looks for "f_user_course".

mulby commented:
:+1:

HassanJaveed84 commented on edx/analytics/tasks/load_internal_reporting_user_course.py:
```diff
@@ -0,0 +1,54 @@
+class LoadInternalReportingUserCourseToWarehouse(WarehouseMixin, VerticaCopyTask):
+    """Loads the course_enrollment Hive table into the f_user_course table in Vertica data warehouse."""
+
+    interval = luigi.DateIntervalParameter()
```
@mulby For this task we use a date on which the course_enrollment hive table was populated. 
Does it make more sense to use a date parameter instead of an interval here ?

brianhw replied:
Interesting.  That's probably a vestige of having the loading tied to the original calculation.  Since we've separated that, we could indeed just pass in a date here.

mulby replied:
Yeah we could, I like that idea.

brianhw commented on edx/analytics/tasks/load_internal_reporting_user_course.py:
```diff
@@ -0,0 +1,54 @@
+    @property
+    def auto_primary_key(self):
+        """Use 'record_number' as primary key to match the schema"""
+        return ('record_number', 'AUTO_INCREMENT')
```
Were we going to leave this in?

mulby replied:
I'm fine either way, we can always remove it later.

HassanJaveed84 commented:
@brianhw @mulby ready for review again. 

- Used date instead of date interval.
- Added read_timeout to test.cfg

I also updated Jenkins job to use the updated parameter.

mulby commented:
:+1:

brianhw commented:
LGTM :+1: .  Squash away!
